.npmignore

```diff
@@ -1,7 +1,40 @@
-/*
-!/addon
-!/app
-!/config
+# compiled output
+/dist/
+/tmp/
+
+# dependencies
+/bower_components/
+
+# misc
+/.bowerrc
+/.editorconfig
+/.ember-cli
+/.env*
+/.eslintcache
+/.eslintignore
+/.eslintrc.js
+/.git/
+/.github/
+/.gitignore
+/.prettierignore
+/.prettierrc.js
+/.template-lintrc.js
+/.travis.yml
+/.watchmanconfig
+/bower.json
 /config/ember-try.js
-!/index.js
-**/.gitkeep
+/CONTRIBUTING.md
+/ember-cli-build.js
+/testem.js
+/tests/
+/yarn-error.log
+/yarn.lock
+.gitkeep
+
+# ember-try
+/.node_modules.ember-try/
+/bower.json.ember-try
+/npm-shrinkwrap.json.ember-try
+/package.json.ember-try
+/package-lock.json.ember-try
+/yarn.lock.ember-try
```

Comment:
Could we revert this file to explicitly include rather than include everything and explicitly exclude?

Reply:
This is not how `.npmignore` [is supposed to be used](https://docs.npmjs.com/cli/v8/using-npm/developers#keeping-files-out-of-your-package). `.npmignore` is a blacklist of things that you do not want to include. If we want to whitelist, we could use the `files`-property in package.json. 

> If, given the structure of your project, you find .npmignore to be a maintenance headache, _you might instead try populating the files property of package.json, which is an array of file or directory names that should be __included__ in your package_. Sometimes manually picking which items to allow is easier to manage than building a block list.

The mechanism that is added here is the default that ember-cli will create for you when you create an add-on. Why would we want to customize that approach to do something different? It does the right thing with this `.npmignore`

Reply:
I really don't mind however you want to do it, if you feel that strongly that you must do this within a JSON file then that's fine by me, the ask here is to stick to the "explicitly include" approach that exists in this package previous to this PR and I don't see that either in the npmignore file or in the package files array here.

> What's the benefit of not following the convention, and customizing that approach to do something different? It does the right thing with this .npmignore

Because ember's convention is "include everything and then explicitly exclude", that's not the convention we want nor the one that exists in the project/package already.

Whilst I really don't mind whether you do this via an npmignore or a files array as you've suggested, I can't see anywhere in either the npmignore that the npmignore file should not be used for specifying which files to ignore, yet it does also suggest that the ember convention of "include everything and explicitly exclude" could be a maintenance headache, from the page linked to:

> If, given the structure of your project, you find .npmignore to be a maintenance headache, you might instead try populating the files property of package.json, which is an array of file or directory names that should be included in your package. Sometimes manually picking which items to allow is easier to manage than building a block list.

Sometimes manually picking which items to allow is easier to manage than building a block list.

The documentation on the package files field also includes the following:

> You can also provide a .npmignore file in the root of your package or in subdirectories, which will keep files from being included.

https://docs.npmjs.com/cli/v8/configuring-npm/package-json#files

Just to summarise, I don't mind either way what you decide to use to "explicitly include" files from a release and judging by the documents linked to here I don't think anyone else does either, just that's how we would like it to work. 

Please could you use either approach to stick to the "explicitly include" approach

Reply:
I pointed out and added a link to where the docs explicitly state that `.npmignore` is for blacklisting and `files` is supposed to be used for whitelisting.

The current approach the add-on takes is going against that recommended approach in the NPM publishing docs, that's the reason I pointed this out. I updated the PR and added a `files`-property to explicitly tell NPM what files to include given you want to use this approach.

Reply:
awesome thankyou!